tests/test_osfs.py

```diff
@@ -152,3 +151,16 @@ def test_validatepath(self):
             with self.assertRaises(errors.InvalidCharsInPath):
                 with self.fs.open("13 – Marked Register.pdf", "wb") as fh:
                     fh.write(b"foo")
+
+    def test_consume_geturl(self):
+        self.fs.create("foo")
+        try:
+            url = self.fs.geturl("foo")
+        except errors.NoURL:
+            self.assertFalse(self.fs.hasurl("foo"))
+        else:
+            self.assertTrue(self.fs.hasurl("foo"))
+
+        # Should not throw an error
+        base_dir = dirname(url)
```

Comment:
here is the test for it : @althonos

Reply:
@chfw Based on the discussion in this PR, I guess this `test_consume_geturl` ought to be using the `purpose="fs"` ?

Reply:
Well, this one is for OSFS. Hence, ‘download’ is assumed.

Reply:
This test will ensure the return value of OSFS.geturl() can be consumed by pyfs2 in all os platforms, especially on windows.

Reply:
As Will said earlier:
"This isn't actually the intent of the geturl method. The URL returned was to be something that could be consumed externally by a browser for example. The URLs you have added are FS URLs which may not be of much use outside of PyFilesystem."

And as I mentioned in another comment (which GitHub may be auto-hiding) file URLs _are_ a perfectly valid thing (see https://blogs.msdn.microsoft.com/ie/2006/12/06/file-uris-in-windows/ ), and may be different to the "FS URLs" which e.g. `open_fs` expects.
Perhaps I wasn't explicit enough, but it looks like perhaps (at least on Windows) `OSFS.geturl()` might be returning "invalid" file URLS? @willmcgugan

Reply:
You have made your point. Let me update the code.